system_metrics_collector/test/system_metrics_collector/test_periodic_measurement_node.cpp

```diff
@@ -120,14 +117,17 @@ class PeriodicMeasurementTestFixure : public ::testing::Test
   }
 
 protected:
+  static constexpr std::chrono::milliseconds INFREQUENT_PUBLISH_PERIOD = 2 *
+    test_constants::TEST_DURATION;
```

Comment:
adding a comment to explain why this value is OK would be helpful, not sure if I understand very well what all those constants mean, and why this is safe and this test won't be flaky.

Reply:
also, it's unclear what's the value here there. Once every 10ms, 10 seconds? INFREQUENT is not really helpful in this context.

Reply:
@thomas-moulard and @dabonnie, can you take a look again?

Reply:
Thanks @mm318 - before merging could you run this with `--gtest_repeat=100000` to make sure it's not flaky?

Reply:
I was able to repeat the test 14000+ times without fail before running into some unrelated RTPS error (like this https://github.com/aliasrobotics/aztarna/issues/41)...

I am confident that the test is not flaky, as we should absolutely not see any publishing if the publishing period is set to *twice* that of the entire allotted test duration, even at the millisecond scale.

If we do see some publishing occur given this much margin (of the entire test duration), I would be concerned about the ROS2 executor.